app/scripts/lib/message-manager.js

```diff
@@ -79,8 +79,8 @@ export default class MessageManager extends EventEmitter {
    *
    */
   addUnapprovedMessageAsync(msgParams, req) {
+    const msgId = this.addUnapprovedMessage(msgParams, req);
```

Comment:
Tests were failing due to race condition, trying to access unapproved list before the Promise even started

Reply:
Huh, interesting. So this situation isn't ideal because if `addUnapprovedMessage` throws, it'll do so synchronously, but if anything inside the Promise throws, it'll reject. This might be confusing if anyone tries to handle errors with `.catch` and this throws and slips past that attempt.

Could you remove the `Promise` wrapper altogether and make this function `async`? Or would that trigger the same issue you encountered here I wonder. Which test was it that was failing?

Reply:
It failed here - https://github.com/MetaMask/metamask-extension/blob/6309d51e359aa0d1788f180fb410e39a1e2bd0d8/app/scripts/metamask-controller.test.js#L857-L864
With `messages` returned being empty.

I don't think this promise can be easily converted into async, since it waits on EventEmitter